Controllers/StaffController.cs

```diff
@@ -12,34 +13,40 @@ namespace stunning_robot_HR.Controllers
 {
     public class StaffController : Controller
     {
-        private readonly stunning_robot_HRContext _context;
-
-        public StaffController(stunning_robot_HRContext context)
+        private stunning_robot_HRContext _context;
+        private  IStaffRepository staffRepository;
+        public StaffController(stunning_robot_HRContext context )
         {
             _context = context;
+            this.staffRepository = new StaffRepository(_context);
         }
-
+        
+        
         // GET: Staff
         public async Task<IActionResult> Index()
         {
-            return View(await _context.Staff.ToListAsync());
+             var staffs =  from s in staffRepository.GetStaff()
+                select s;
+             return View(staffs);
         }
-
-        // GET: Staff/Details/5
-        public async Task<IActionResult> Details(int? id)
+        
+        public ViewResult SearchStaff(string searchString)
         {
-            if (id == null)
-            {
-                return NotFound();
-            }
-
-            var staff = await _context.Staff
-                .FirstOrDefaultAsync(m => m.StaffId == id);
-            if (staff == null)
+            var staffs = from s in staffRepository.GetStaff()
+                select s;
+             
+            if (!String.IsNullOrEmpty(searchString))
             {
-                return NotFound();
+                staffs = staffs.Where(s => s.FullName.Contains(searchString));
```

Comment:
This guy belongs in your repository for Staff, likely a method called GetStaffByFullName(string searchCriteria)